io/include/pcl/io/pcd_io.h

```diff
@@ -600,6 +600,33 @@ namespace pcl
       resetLockingPermissions (const std::string &file_name,
                                boost::interprocess::file_lock &lock);
 
+      /** \brief potential errors in the size of the cloud to be written to disk
+       * success = no discrepancy detected)
+       * empty_cloud = the input point cloud has zero points)
+       * size_mismatch = the number of points in the cloud do not match height*width)
+       */
+       enum
+      {
+        success = 0,
+        empty_cloud = 1, 
+        size_mismatch = 2
+      };
```

Comment:
Have a look at enum classes. Otherwise this will leak into the pcl namespace. Also use full CAPITAL_CASE for constants.
https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum

Reply:
With this approach, the return type of the checkSize() method should be switched from `int` to the enum class type right? something like `pcl::PCDWriter::cloud_status`?

Reply:
SO has an [answer](https://stackoverflow.com/a/9883421) that allows downstream users to adapt to API change